File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;

    function dfs(i, total) {
        if (i < numbers.length) {
            dfs(i+1, total + numbers[i]);
            dfs(i+1, total - numbers[i]);
        } else {
            if(total === target) answer++
        }
    }
    
    dfs(0,0);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    queue = [[numbers[0],0], [-1 * numbers[0],0]]
    n = len(numbers)

    while queue:
        cur, i = queue.pop()
        i += 1
        if i < n:
            queue.append([cur+numbers[i], i])
            queue.append([cur-numbers[i], i])
        else:
            if cur == target: answer += 1
            
    return answer